libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1065,6 +1065,46 @@ string YulUtilFunctions::clearStorageArrayFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::clearStorageStructFunction(StructType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+
+	string functionName = "clear_struct_storage_" + _type.identifier();
+
+	return m_functionCollector.createFunction(functionName, [&] {
+		auto const& structType = dynamic_cast<StructType const&>(_type);
+		MemberList::MemberMap structMembers = structType.nativeMembers(nullptr);
+		vector<map<string, string>> memberSetValues(structMembers.size());
+
+		for (size_t i = 0; i < structMembers.size(); ++i)
+		{
+			auto const& [memberSlotDiff, memberStorageOffset] = structType.storageOffsetsOfMember(structMembers[i].name);
+
+			memberSetValues[i]["clearMember"] = Whiskers(R"(
+			{
+				<setZero>(add(slot, <memberSlotDiff>), <memberStorageOffset>)
+			}
+			)")
+			("setZero", storageSetToZeroFunction(*structMembers[i].type))
+			("memberSlotDiff",  memberSlotDiff.str())
+			("memberStorageOffset", to_string(memberStorageOffset))
+			.render();
+		}
```

Comment:
Do we really need to do this member by member? We said that we could have more optimised versions later, but I think here the "optimised version" would actually also be the simpler version...

In storage we can assume that the "zero value" of every type is *actually* plain zero - and structs will always start at full slots and are padded to full slots at the end, aren't they?

If that's true, then can't we just loop over ``structType.storageSize()`` many storage slots starting from ``slot`` and just assign all of them plainly to zero?

Reply:
I would be fine by setting all of them to zeros. Only potential problem I see is if struct contains reference types, for example array. If we do it this way, storage that was taken by array will not be cleared. Or it is not important since we just set `ref` to zero and we do not care about what stayed at the place where it was pointing...

Reply:
Hm, yes, you're right, I hadn't thought about that properly - any dynamically encoded reference type has to be handled differently... *But*: all those will still occupy a full slot...
So an "optimal" version of this would probably first iterate over all reference type members and call the recursive clearing function for them - and then only after that's done clear all slots in the struct itself.

Reply:
Ok, I will update to "optimal version" now